### PR TITLE
fix(color-control): Propagate colorMode

### PR DESCRIPTION
**Pull request by kennylevinsen**
We need to set the initial colorMode to a supported value, as this will be used for future colot conversions internally in matter.js. Set it to the color mode used by Home Assistant, while falling back to any valid value.

Fixes: #225

### CODE REVIEW

**Comment by Apollon77** [comment]
The question is if there is any place in the HA data to know the current color mode because it should then be set to this instead of to your proposal because also this would be wrong. And when this way overrides any value that was set before on start ,.... so also not the right thing.

@t0bst4r maybe you should adjust in general to ggive defaults on endpoint initialization as planned - then potentially persisted values (like this one) are used from storage rather than default

**Comment by t0bst4r** [comment]
Yes, there is a [property](https://developers.home-assistant.io/docs/core/entity/light/#properties) to determine the color mode, but i just forgot it here 😆 

instead of hard coding "color temperature" as default color mode, we can parse it to the right one.

HINT: we only support `hue and saturation` and `color temp`, but not `xy` or `rgb` or whatever else, because
1. HA has way more modes than matter
2. most lights have a `hs_color` property even if they don't support HS
3. if the `hs_color` is missing we can convert each HA **color** mode to HS
4. HA accepts HS even if the light doesn't - HA converts it for us.

So basically we can just map it like this (pseudo code):
```
if this.features.colorTemperature && entity.colorMode == "temp"
  matter.colorMode = "temperature"
else if this.features.hue_saturation
  matter.colorMode = "hs"
else
  // somehow handle invalid value
```

**Comment by kennylevinsen** [comment]
So I made some changes:
1. `colorMode` is now updated on change to supported values, under the assumption that if a HA light reports a particular color mode that it will also report the relevant `current*` values for conversions to be valid.
2. If the current HA color mode is not supported, a check in `initialize` will pick a default once.
3. If no supported features were enabled, it is unable to pick a color mode and throws an error. Not sure what we should do here.

I assume that the `features` bitmask will be set according to some negotiation with the matter controller and aren't just the features we statically requested at the top.

**Comment by t0bst4r** [comment]
I updated your pr:
- ColorControlServer is always created with `HueSaturation` and/or `ColorTemperature`. It can be created with both, but never without any.
- XY feature is never used, we always use HS - even for XY-only entities.
- We don't need choose the default in initialize, when we can do it in update. Update is called by initialize, too, and update will only set it, if current state is different - which is once during initialization and everytime it CHANGES.

File: packages/backend/src/matter/behaviors/color-control-server.ts [hunk 1]
```diff
@@ -174,6 +142,28 @@ export class ColorControlServerBase extends FeaturedBase {
...
+    // This cluster is only used with HueSaturation, ColorTemperature or Both.
+    // It is never used without any of them.
+    if (this.features.colorTemperature && this.features.hueSaturation) {
+      if (mode === LightDeviceColorMode.COLOR_TEMP) {
+        return ColorControl.ColorMode.ColorTemperatureMireds;
+      } else {
+        return ColorControl.ColorMode.CurrentHueAndCurrentSaturation;
+      }
+    } else if (this.features.colorTemperature) {
+      return ColorControl.ColorMode.ColorTemperatureMireds;
```
Comment (kennylevinsen) [inline packages/backend/src/matter/behaviors/color-control-server.ts:158]: The idea of the previous setup was to let matter.js state be (which will internally transition around on varios commansd) if the light entity wasn't in a supported color mode, but I suppose in retrospect that it doesn't make much sense as we'd still be messing with the `current*` state anyway...
